ui/src/main/kotlin/org/jetbrains/jewel/ui/component/TabStrip.kt

```diff
@@ -91,23 +89,71 @@ public sealed class TabData {
     @GenerateDataFunctions
     public class Default(
         override val selected: Boolean,
-        override val label: String,
+        override val content: @Composable (tabState: InteractiveComponentState) -> Unit,
         override val icon: Painter? = null,
         override val closable: Boolean = true,
         override val onClose: () -> Unit = {},
         override val onClick: () -> Unit = {},
-    ) : TabData()
+    ) : TabData() {
+
+        @Deprecated(
+            "Use the primary constructor instead",
+            ReplaceWith(
+                "Default(selected = selected, content = { Text(text) }, icon = icon, closable = closable, onClose = onClose, onClick = onClick)",
+                "org.jetbrains.jewel.ui.component.TabData.Default",
+            ),
+        )
+        public constructor(
+            selected: Boolean,
+            text: String,
+            icon: Painter? = null,
+            closable: Boolean = true,
+            onClose: () -> Unit,
+            onClick: () -> Unit,
+        ) : this(
+            selected = selected,
+            content = { Text(text) },
+            icon = icon,
+            closable = closable,
+            onClose = onClose,
+            onClick = onClick,
+        )
+    }
 
     @Immutable
     @GenerateDataFunctions
     public class Editor(
         override val selected: Boolean,
-        override val label: String,
+        override val content: @Composable (tabState: InteractiveComponentState) -> Unit,
         override val icon: Painter? = null,
         override val closable: Boolean = true,
         override val onClose: () -> Unit = {},
         override val onClick: () -> Unit = {},
-    ) : TabData()
+    ) : TabData() {
+
+        @Deprecated(
+            "Use the primary constructor instead",
+            ReplaceWith(
+                "Editor(selected = selected, content = { Text(text) }, icon = icon, closable = closable, onClose = onClose, onClick = onClick)",
+                "org.jetbrains.jewel.ui.component.TabData.Default",
```

Comment:
```suggestion
                "org.jetbrains.jewel.ui.component.TabData.Editor",
```

Re-read when you copy paste, please 😁